cmd/kube-proxy/app/init_windows.go

```diff
@@ -0,0 +1,54 @@
+// +build windows
+
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"os"
+
+	"k8s.io/kubernetes/pkg/windows/service"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+const (
+	serviceName        = "kube-proxy"
+	serviceDisplayName = "Kubernetes network proxy"
+)
+
+func initForOS() {
+	// On Windows, this may be launching as a service or with an option to
+	// register the service.
+	stop, runAsService, err := service.InitService()
+	if err != nil {
+		glog.Fatal(err)
```

Comment:
This function should return an error instead of calling `glog.Fatal`. Same for the kubelet version.

Reply:
Would there be some way to make this reusable in a separate package, so we don't have to duplicate it (`initForOS` and `addOSFlags`) in each component?

Reply:
Sure, I will update the function to return the error instead of calling ```glog.Fatal```.

In the first version there was a separate package which contained the code. I changed it since @mtaufen wanted this to be inside the ```options```  package from kubelet with the other flag installers. For more details you can view the previous discussion: https://github.com/kubernetes/kubernetes/pull/60144#discussion_r170788021